.github/workflows/build-source-binary.yml

```diff
@@ -0,0 +1,65 @@
+##
+## Builds binary assets of stacks-blockchain and creates a named tag github (draft) release
+##
+
+name: Build Distributable Assets
+
+# Only run when:
+#   - manually triggered via the ci.yml workflow with a provided input tag
+
+on:
+  workflow_call:
+    inputs:
+      tag:
+        description: "Tag name of this release (x.y.z)"
+        required: true
+        type: string
+      parallel_jobs:
+        description: "Number of parallel binary builds"
+        required: false
+        type: number
+        default: 4
+      arch:
+        description: "Stringified JSON object listing of platform matrix"
+        required: true
+        type: string
+
+jobs:
+  artifact:
+    if: ${{ inputs.tag != '' }}
+    name: Create Artifacts
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: ${{ inputs.parallel_jobs}}
+      matrix:
+        platform: ${{fromJson(inputs.arch)}}
```

Comment:
Clever!

Reply:
thanks! i tried to find ways to prevent hard-coding values in the workflows or steps wherever possible.
You'll also notice I changed checking for  `master` branch to checking if the branch is the `default` - anticipating that we may move to `main` at some point, so the workflow would still be valid.  

Where i failed is that with this PR there are still 2 separate workflows for building a debian and alpine image, but the steps are nearly identical (i ran into some issues with differences in tagging the images so i moved on since the current logic was working well enough). 

So, not the ideal i set out with but with more time i'm sure it can be solved and we can remove some lines from the repo (already thinking about how it can be done, but better to get this PR merged and addressed later was the thought process).